dependencies/mqtt/dialer_test.go

```diff
@@ -4,9 +4,33 @@ import (
 	"context"
 	"testing"
 
+	"github.com/influxdata/flux/dependencies/feature"
 	"github.com/influxdata/flux/dependencies/mqtt"
+	"github.com/influxdata/flux/dependency"
+	"github.com/influxdata/flux/execute/executetest"
 )
 
+type MockDialer struct {
```

Comment:
I like this mock. It looks closer to what I expect from mocks than other mocks I've seen in our various go codebases.